### PR TITLE
Adding mac and windwos unit test for main branch

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

### CODE REVIEW

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml [hunk 1]
```diff
@@ -52,6 +52,106 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )psi-kubernetes-integration-e2e,?($|\s.*)
+  - agent: kubernetes
+    branches:
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:56]: Why `always_run:` is not there 🤔  . Did you run `make jobs` ?

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml (hunk 1 above)
Comment (anandrkskd) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:56]: yes, we don't want to run tests on docs PR right, so we need to remove `always_run` and added `run_if_changed`.

Comment (prietyc123) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:56]: ok, could you please remove from other places too. Just to be in sync.

Comment (anandrkskd) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:56]: I didn't understand by the term "other place"

Comment (prietyc123) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:56]: I am talking about other jobs. For ex https://github.com/openshift/release/pull/18247/files#diff-2a14d19c92854dd1baf93157739f79151bf0d5770aee591a1f0c41bf819863c6R4

Comment (anandrkskd) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:56]: which one? I guess all of them have `run_if_changed`

**Pull request merged**